src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -110,14 +118,27 @@ internal IChecksummedPersistentStorage GetStorageWorker(Solution solution)
             }
         }
 
-        private static bool DatabaseSupported(Solution solution, bool checkBranchId)
+        private string? TryGetWorkingFolder(Workspace workspace, SolutionKey solutionKey, Solution? solutionOpt)
+        {
+            // First, see if we have the new API that just operates on a Workspace/Key.  If so, use that.
+            if (_locationService is IPersistentStorageLocationService2 locationService2)
+                return locationService2.TryGetStorageLocation(workspace, solutionKey);
+
+            // Otherwise, use the existing API.  However, that API only works if we have a full Solution to pass it.
+            if (solutionOpt == null)
+                return null;
+
+            return _locationService.TryGetStorageLocation(solutionOpt);
```

Comment:
@Therzok  Do you guys implement your own IPersistentStorageLocationService?  Or do you just subclass our DefaultPersistentStorageLocationService?  If only the latter, we can just change the api of IPersistentStorageLocationService.  I wasn't sure, so i went the approach of introducing a new optional IPersistentStorageLocationService2 api.